lib/tapioca/compilers/dsl/active_record_scope.rb

```diff
@@ -0,0 +1,92 @@
+# typed: true
+# frozen_string_literal: true
+
+require "parlour"
+
+begin
+  require "active_record"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `Tapioca::Compilers::Dsl::ActiveRecordScope` decorates RBI files for subclasses of
+      # `ActiveRecord::Base` which declare `scope` fields
+      # (see https://api.rubyonrails.org).
+      #
+      # For example, with the following `ActiveRecord::Base` subclass:
+      #
+      # ~~~rb
+      # class Post < ApplicationRecord
+      # end
+      # ~~~
+      #
+      # this generator will produce the RBI file `post.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # post.rbi
+      # # typed: true
+      # class Post
+      # end
+      # ~~~
+      class ActiveRecordScope < Base
+        extend T::Sig
+
+        sig do
+          override.params(
+            root: Parlour::RbiGenerator::Namespace,
+            constant: T.class_of(::ActiveRecord::Base)
+          ).void
+        end
+        def decorate(root, constant)
+          scopes = constant.send(:generated_relation_methods).instance_methods(false)
```

Comment:
Let's call this `scope_methods` instead.

Reply:
I did iterate through scopes before calling the `generate_instance_methods`, therefore only sending the scope_method name as parameter.

Reply:
Thanks, that change makes sense. However, this variable is still not "scopes", it is just a list of method names that corresponds to defined scopes. So a better name would be `scope_method_names`.